### ISSUE TITLE
[New Concept Exercise] : other-comprehensions

### ISSUE THREAD

**Issue by BethanyG** [exercism/python#2294]
This issue describes how to implement the `other-comprehensions` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [Contributing to Exercism](https://exercism.org/docs/building) | [Exercism and GitHub](https://exercism.org/docs/building/github) | [Contributor Pull Request Guide](https://exercism.org/docs/building/github/contributors-pull-request-guide)
- [What are those Weird Task Tags about?](https://exercism.org/docs/building/product/tasks)
- [Building Language Tracks: An Overview](https://exercism.org/docs/building/tracks) 
- [What are Concepts?](https://exercism.org/docs/building/tracks/concepts)
- [Concept Exercise Specifications](https://exercism.org/docs/building/tracks/concept-exercises)
- [Concept Specifications](https://exercism.org/docs/building/tracks/concepts) 
- [Exercism Formatting and Style Guide](https://exercism.org/docs/building/markdown/style-guide)
- [Exercism Markdown Specification](https://exercism.org/docs/building/markdown/markdown)
- [Reputation](https://exercism.org/docs/using/product/reputation)  

## Goal

The goal of this exercise is to teach the syntax and variants of `set comprehensions` and `dict comprehensions` in Python. 

## Learning objectives

-  Understand how `set` and `dict`  comprehensions relate to their underlying data structures and the `loop` + `append` method of creating/computing them.
-  Create a `dict` comprehension from a `loop` + `append`
-  Create a `set` comprehension from a `loop` + `append`
-  Create a `dict` comprehension from `Lists`,  `Sets`,  `Tuples`, or other `iterables` (_such as `zip()` or `dict.items()`_)
-  Create a `set` comprehension from `Lists`,  `Sets`,  `Tuples`, or other `iterables` (_such as `zip()` or `dict.items()`_)
-  Use one or more conditions/operators/methods to filter comprehension inputs
-  Use methods or logic to format the elements (output members) of the comprehension
-  Create a _nested comprehension_  (of either flavor)
-  Create a _nested comprehension_ (of either flavor) with one or more formatting or filtering conditions

## Out of scope

- Memory and performance characteristics and optimizations
- `generators` and `generator expressions` in `other comprehensions`
-  using the data structures in `collections` in combination with, or as part of a `set` or `dict` comprehension.
-  Using the `assignment expression` (_walrus operator_) with either flavor of comprehension.

## Concepts

-  `dict-comprehensions`
-   `set-comprehensions`
-   `comprehension syntax`

## Prerequisites

- `basics`
-  `bools`
- `conditionals`
- `comparisons`
- `loops`
- `iteration`

## Resources to refer to

-  [List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- [Nested List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#nested-list-comprehensions)
- [Comprehending Python's Comprehensions (Dan Bader)](https://dbader.org/blog/list-dict-set-comprehensions-in-python)
- [List and Dict Comprehensions in Python (Timothy Bramlett)](https://timothybramlett.com/List_and_Dict_Comprehensions_in_Python.html)

### Hints

- `List Comprehensions` section of the Python docs tutorial:  [List Comprehensions](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- This animated GIF from Trey Hunner:  [List Comprehensions: The Movie](https://treyhunner.com/images/list-comprehension-condition.gif)

### After

-  `comprehension syntax` for other data structures such as `sets` and `dictionaries`
- `generators` and `generator expressions`
- `generators` and `generator expressions` in `list comprehensions`

## Representer

No changes required.

## Analyzer

No changes rquired.

## Implementing

- Tests should be written using unittest.TestCase, and the test file named list_comprehensions_test.py.
- [How to Implement a Concept Exercise in Python](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md)
- [make-concept-exercise Utility](https://github.com/exercism/v3/tree/master/languages/python/bin)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
